app/models/conversation.rb

```diff
@@ -20,7 +22,35 @@ class Conversation < ApplicationRecord
     order(has_unread_jobseeker_messages: :desc, last_message_at: :desc)
   }
 
+  include PgSearch::Model
+
+  pg_search_scope :search_by_keyword,
+                  against: [:searchable_content],
+                  using: {
+                    tsearch: {
+                      prefix: true,
+                      any_word: true,
+                      dictionary: "english",
+                    },
+                  }
+
   def has_unread_messages_for_publishers?
     messages.any? { |msg| msg.is_a?(JobseekerMessage) && msg.unread? }
   end
+
+  def update_searchable_content
+    update!(searchable_content: generate_searchable_content)
+  end
+
+  def generate_searchable_content
+    message_content = messages.filter_map do |message|
+      message.content.to_plain_text
+    end
+
+    Search::Postgres::TsvectorGenerator.new(
+      a: [job_application.vacancy.job_title],       # Job title (highest weight)
+      b: [job_application.name],                    # Candidate name (mid weight)
+      d: message_content, # Message content (lower weight)
```

Comment:
Nitpick 💄  non-blocking
```suggestion
    Search::Postgres::TsvectorGenerator.new(
      a: [job_application.vacancy.job_title], # Job title (highest weight)
      b: [job_application.name],              # Candidate name (mid weight)
      d: message_content,                     # Message content (lower weight)
```

Reply:
Thanks for the comments on the weighting 👏